astropy/wcs/utils.py

```diff
@@ -118,6 +158,62 @@ def _wcs_to_celestial_frame_builtin(wcs):
                 representation_type=SphericalRepresentation,
                 obstime=wcs.wcs.dateobs or None,
             )
+        elif xcoord[2:4] in ("LN", "LT") and "H" not in xcoord and "CR" not in xcoord:
+            # Coordinates on a planetary body, as defined in
+            # https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2018EA000388
+
+            object_name = SOLAR_SYSTEM_OBJ_DICT.get(xcoord[:2])
+            if object_name is None:
+                raise KeyError(f"unknown solar system object abbreviation {xcoord[:2]}")
+
+            a_radius = wcs.wcs.aux.a_radius
+            b_radius = wcs.wcs.aux.b_radius
+            c_radius = wcs.wcs.aux.c_radius
+            if "bodycentric" in wcs.wcs.name.lower():
+                baserepresentation = BaseBodycentricRepresentation
+                representation_type_name = "BodycentricRepresentation"
+            else:
+                baserepresentation = BaseGeodeticRepresentation
+                representation_type_name = "GeodeticRepresentation"
+
+            if a_radius == b_radius:
+                equatorial_radius = a_radius * u.m
+                flattening = (a_radius - c_radius) / a_radius
+            else:
+                raise NotImplementedError(
+                    "triaxial systems are not supported at this time."
+                )
+
+            # create a new representation class
+            representation_type = solar_system_body_representation_type(
+                SOLAR_SYSTEM_OBJ_DICT.get(xcoord[:2]),
+                baserepresentation,
+                equatorial_radius,
+                flattening,
+            )
+            if representation_type is None:
```

Comment:
This whole clause is no longer needed - the above function call will either create the representation or return the cached function!

Indeed, this is why there is no coverage of these lines.